### ISSUE TITLE
Provide example usage of signing email

### ISSUE THREAD

**Issue by vdot0x23** [emersion/go-pgpmail#9]
@jbjjbjjbj and I tried the snippet below and Evolution said the resulting signature was BAD (i.e. the signature did not match the message.
```go


var h textproto.Header
h.Add("From", n.sender)
h.Add("To", recipient)
h.Add("Subject", subject)
var hEmpty textproto.Header
wcSigned, err := pgpmail.Sign(wc, h, hEmpty, n.secretSigningKey, nil)
if err != nil {
  return err
}
_, err = fmt.Fprintf(wcSigned, body)
if err != nil {
  return err
}

err = wcSigned.Close()
if err != nil {
  return err
}
```

Resulting email is attached (renamed from `.eml` to `.txt`), please let me know if you need more information, thank you :)

[badsig.txt](https://github.com/emersion/go-pgpmail/files/4810361/badsig.txt)

**Comment by emersion** [comment]
The GoDoc page has some examples. see https://godoc.org/github.com/emersion/go-pgpmail#example-Encrypt

**Comment by grmat** [comment]
It's not obvious to use for me either. Using `Sign()` similarly to `Encrypt()` results in header weirdness and the signature will match an empty string, not the written content.

**Comment by emersion** [comment]
Fixed `Sign` in 24b3549e6ef615c9f8d2fa0f5fab33c972d62c5c and 1c654891dbd859fc348f9eaa19ab9118fe0ed7c6, added an example in 8f265f648e48419cb6e48c32fb1b8dc8410aa77a.
